### PR TITLE
e2e-test: Improve test coverage for agent-protocol-forwarder in CAA

### PR DESCRIPTION
**Pull request by ANJANA-A-R-K**
Enhance the test suite for the agent-protocol-forwarder component by adding new test cases across 5 test functions. The improvements include comprehensive tests for Config.Setup() method covering all command-line flag combinations, TLS configuration scenarios (enabled/disabled/skip-verify), edge cases for config file loading (empty files, null values, extra fields, permission errors), host interface configuration, default value handling, and error scenarios (missing files, invalid JSON).

```
go test -v -cover -tags "aws,ibmcloud,libvirt,azure"
go: downloading github.com/coreos/go-systemd v0.0.0-20190719114852-fd7a80b32e1f
go: downloading golang.org/x/sys v0.41.0
go: downloading github.com/kata-containers/kata-containers/src/runtime v0.0.0-20260317012958-660e3bb6535b
go: downloading google.golang.org/protobuf v1.36.11
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20251202230838-ff82c1b0f217
go: downloading google.golang.org/grpc v1.79.3
go: downloading google.golang.org/genproto v0.0.0-20250603155806-513f23925822
go: downloading github.com/coreos/go-iptables v0.6.0
go: downloading github.com/vishvananda/netlink v1.3.1
=== RUN   TestLoad
=== RUN   TestLoad/missing_file
=== RUN   TestLoad/invalid_json
=== RUN   TestLoad/valid_json
--- PASS: TestLoad (0.00s)
    --- PASS: TestLoad/missing_file (0.00s)
    --- PASS: TestLoad/invalid_json (0.00s)
    --- PASS: TestLoad/valid_json (0.00s)
=== RUN   TestConfigSetup
=== RUN   TestConfigSetup/valid_config_with_all_flags
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/config_with_disable-tls_flag
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/config_with_tls-skip-verify_flag
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/version_flag_exits_with_code_0
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/missing_config_file
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/invalid_config_file
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetup/default_values
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
--- PASS: TestConfigSetup (0.00s)
    --- PASS: TestConfigSetup/valid_config_with_all_flags (0.00s)
    --- PASS: TestConfigSetup/config_with_disable-tls_flag (0.00s)
    --- PASS: TestConfigSetup/config_with_tls-skip-verify_flag (0.00s)
    --- PASS: TestConfigSetup/version_flag_exits_with_code_0 (0.00s)
    --- PASS: TestConfigSetup/missing_config_file (0.00s)
    --- PASS: TestConfigSetup/invalid_config_file (0.00s)
    --- PASS: TestConfigSetup/default_values (0.00s)
=== RUN   TestConfigSetupTLSConfiguration
=== RUN   TestConfigSetupTLSConfiguration/TLS_config_with_all_cert_files
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetupTLSConfiguration/TLS_config_with_partial_cert_files
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
--- PASS: TestConfigSetupTLSConfiguration (0.00s)
    --- PASS: TestConfigSetupTLSConfiguration/TLS_config_with_all_cert_files (0.00s)
    --- PASS: TestConfigSetupTLSConfiguration/TLS_config_with_partial_cert_files (0.00s)
=== RUN   TestLoadEdgeCases
=== RUN   TestLoadEdgeCases/empty_config_file
=== RUN   TestLoadEdgeCases/config_with_extra_fields
=== RUN   TestLoadEdgeCases/config_with_null_values
=== RUN   TestLoadEdgeCases/config_file_with_wrong_permissions
    main_test.go:477: skipping test when running as root
--- PASS: TestLoadEdgeCases (0.00s)
    --- PASS: TestLoadEdgeCases/empty_config_file (0.00s)
    --- PASS: TestLoadEdgeCases/config_with_extra_fields (0.00s)
    --- PASS: TestLoadEdgeCases/config_with_null_values (0.00s)
    --- SKIP: TestLoadEdgeCases/config_file_with_wrong_permissions (0.00s)
=== RUN   TestConfigSetupWithDifferentInterfaces
=== RUN   TestConfigSetupWithDifferentInterfaces/with_host_interface_specified
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
=== RUN   TestConfigSetupWithDifferentInterfaces/without_host_interface
agent-protocol-forwarder version unknown
  commit: unknown
  go: go1.25.8 (Red Hat 1.25.8-1.el9_7)
--- PASS: TestConfigSetupWithDifferentInterfaces (0.00s)
    --- PASS: TestConfigSetupWithDifferentInterfaces/with_host_interface_specified (0.00s)
    --- PASS: TestConfigSetupWithDifferentInterfaces/without_host_interface (0.00s)
PASS
coverage: 76.9% of statements
ok  	github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/cmd/agent-protocol-forwarder	0.028s

```

### CODE REVIEW

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/cmd"
+	daemon "github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/forwarder"
+)
+
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go:13]: Would a helper function e.g.
```
const testConfigFileMode = 0o600

// setupTestConfig creates a temporary config file and sets up os.Args for testing.
// Returns the config path and a cleanup function to restore os.Args.
func setupTestConfig(t *testing.T, configContent string, extraArgs ...string) (string, func()) {
	t.Helper()
	tmpDir := t.TempDir()
	configPath := filepath.Join(tmpDir, "config.json")
	
	if err := os.WriteFile(configPath, []byte(configContent), testConfigFileMode); err != nil {
		t.Fatalf("failed to write config file: %v", err)
	}
	
	oldArgs := os.Args
	os.Args = append([]string{"agent-protocol-forwarder", "-config", configPath}, extraArgs...)
	
	return configPath, func() { os.Args = oldArgs }
}
```
help to reduce the boilerplate and duplication here?

Comment (ANJANA-A-R-K) [reply src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go:13]: Sure, changed this!

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+	os.Args = append([]string{"agent-protocol-forwarder", "-config", configPath}, extraArgs...)
+
+	return configPath, func() { os.Args = oldArgs }
+}
+
+func TestLoad(t *testing.T) {
+	t.Run("missing file", func(t *testing.T) {
+		var cfg daemon.Config
+
+		err := load(filepath.Join(t.TempDir(), "missing.json"), &cfg)
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go]: nit: Splitting this up might improve readability?

```suggestion
		path := filepath.Join(t.TempDir(), "missing.json")
	    err := load(path, &cfg)
```

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+		if err == nil {
+			t.Fatal("expected error, got nil")
+		}
+		if !strings.Contains(err.Error(), "failed to open") {
+			t.Fatalf("expected open error, got %v", err)
+		}
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go]: Could we make this more concise and readable with testify/assert e.g.
```suggestion
        assert.ErrorIs(t, err, os.ErrNotExist)
		assert.Contains(t, err.Error(), "failed to open")
```

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+	})
+
+	t.Run("invalid json", func(t *testing.T) {
+		path := filepath.Join(t.TempDir(), "config.json")
+
+		if err := os.WriteFile(path, []byte("{invalid"), testConfigFileMode); err != nil {
+			t.Fatalf("failed to write config file: %v", err)
+		}
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go]:
```suggestion
    require.NoError(t, os.WriteFile(path, []byte("{invalid"), testConfigFileMode))
```

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+
+		var cfg daemon.Config
+
+		err := load(path, &cfg)
+		if err == nil {
+			t.Fatal("expected error, got nil")
+		}
+		if !strings.Contains(err.Error(), "failed to decode") {
+			t.Fatalf("expected decode error, got %v", err)
+		}
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go]:
```suggestion
assert.ErrorContains(t, err, "failed to decode")
```

File: src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+		if cfg.TLSServerKey != "server-key" {
+			t.Fatalf("expected tls server key %q, got %q", "server-key", cfg.TLSServerKey)
+		}
+		if cfg.TLSServerCert != "server-cert" {
+			t.Fatalf("expected tls server cert %q, got %q", "server-cert", cfg.TLSServerCert)
+		}
+		if cfg.TLSClientCA != "client-ca" {
+			t.Fatalf("expected tls client ca %q, got %q", "client-ca", cfg.TLSClientCA)
+		}
```
Comment (stevenhorsman) [inline src/cloud-api-adaptor/cmd/agent-protocol-forwarder/main_test.go]: An expected object helps the readability here too.
```suggestion
		expected := daemon.Config{
			PodNamespace:  "test-namespace",
			PodName:       "test-pod",
			TLSServerKey:  "server-key",
			TLSServerCert: "server-cert",
			TLSClientCA:   "client-ca",
		}

		var cfg daemon.Config
		err := load(path, &cfg)

		assert.NoError(t, err)
		assert.Equal(t, expected, cfg)
```

**Review by stevenhorsman — COMMENTED** [review]
I've given some suggestions around using testify framework to help improve the readability that can be rolled out to the other tests.

**Review by stevenhorsman — APPROVED** [review]
Looks cleaner to me now. Thanks for the update

**Review by mkulke — APPROVED** [review]
